### ISSUE TITLE
Can't add product to cart with customizable options (type checkbox) through REST API

### ISSUE THREAD

**Issue by gekamikh** [magento/magento2#23863]
### Preconditions (*)

1. Magento version: 2.3.1
2. PHP version: 7.2.12

### Steps to reproduce (*)

1. Try to add product to guest cart using API endpoint: `POST /V1/guest-carts/{cartId}/items`
Body:
```
{
    "cartItem": {
        "sku": "sku",
        "qty": 1,
        "quote_id": "quoteId",
        "product_option": {
            "extension_attributes": {
                "custom_options": [
                    {
                        "option_id": "12",
                        "option_value": "20,21,22"
                    }
                ]
            }
        }
    }
}
```

### Expected result (*)

1. Product added to cart

### Actual result (*)

1. Error:
```
{
    "message": "The product's required option(s) weren't entered. Make sure the options are entered and try again."
}
```

Error thrown from:
https://github.com/magento/magento2/blob/2.3.1/app/code/Magento/Catalog/Model/Product/Option/Type/Select.php#L77

Potentially this commit 64ee8e9 broke functionality, because if change `0` to `1` as it was before this fix, everything works. Product added to the cart

**Comment by Den4ik** [comment]
@gekamikh I have checked and it seems you were use wrong params in query.
You should to use cartId instead of quoteId in request body `"quote_id": "cartId"`

**Comment by gekamikh** [comment]
@Den4ik, yes I use `cartId`. Before add product to cart I create it using API route `POST /V1/guest-carts`. After that I get cartId from response and use it for all requests to cart endpoints. So it's not a solution.

Also when I try to add product with customizable option `radio` everything works. So, please, check again

**Comment by Den4ik** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @Den4ik. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, here is your [Magento instance](https://i-23863-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23863-2-3-develop.instances.magento-community.engineering/admin](https://i-23863-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
@Den4ik Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by Den4ik** [comment]
@gekamikh Yep, issue reproduced for multiselect custom option

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @Den4ik 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-18563` were created

**Issue Available**: @Den4ik, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sidolov** [comment]
Hi @gekamikh. Thank you for your report. 
The issue has been fixed in magento/magento2#23871 by @Den4ik in 2.3-develop branch
Related commit(s):
 - [1acde36b19b177b58af916557a3d41dfe0612cc0](https://github.com/magento/magento2/commit/1acde36b19b177b58af916557a3d41dfe0612cc0)

The fix will be available with the upcoming 2.3.3 release.

**Comment by derappelt** [comment]
> Hi @gekamikh. Thank you for your report.
> The issue has been fixed in #23871 by @Den4ik in 2.3-develop branch
> Related commit(s):
> 
> * [1acde36b19b177b58af916557a3d41dfe0612cc0](https://github.com/magento/magento2/commit/1acde36b19b177b58af916557a3d41dfe0612cc0)
> 
> The fix will be available with the upcoming 2.3.3 release.

It is **NOT** fixed in 2.3.3 
Took me hours to find out since I believed you and looked for the error elsewhere.
The [1acde36b19b177b58af916557a3d41dfe0612cc0](https://github.com/magento/magento2/commit/1acde36b19b177b58af916557a3d41dfe0612cc0) solves the issue but it's not in 2.3.3 so this issue is still open
